mlflow/projects/__init__.py

```diff
@@ -38,6 +39,9 @@ def _run(uri, entry_point="main", version=None, parameters=None, experiment_id=N
     Helper that delegates to the project-running method corresponding to the passed-in mode.
     Returns a ``SubmittedRun`` corresponding to the project run.
     """
+    if mode == "databricks":
+        mlflow.projects.databricks.before_run_validations(mlflow.get_tracking_uri(), cluster_spec)
+
     exp_id = experiment_id or _get_experiment_id()
```

Comment:
How do you feel about moving the project run creation logic into a separate method: `_create_project_run()`? We can then move the new `mode` check into the condition on line 67 and call `_create_project_run()` in both the if/elif branches.